CRM/Utils/System/Drupal.php

```diff
@@ -921,4 +921,14 @@ public function theme(&$content, $print = FALSE, $maintenance = FALSE) {
     }
   }
 
+  /**
+   * @inheritdoc
+   */
+  public function ipAddress():?string {
+    // Drupal function handles the server being behind a proxy securely. We
+    // still have legacy ipn methods that reach this point without bootstrapping
+    // hence the check that the fn exists.
+    return ip_address();
```

Comment:
Like backdrop, maybe return function_exists('ip_address') ? ip_address() : ($_SERVER['REMOTE_ADDR'] ?? NULL);

Reply:
Thanks @adixon. Good point, I'll make the change.

Reply:
I added it to the Drupal8 method as well. The same conditions would apply I believe: no bootstrap, no `Drupal` class.